SPECS/python-rich/0001-Skip-tests-for-mariner.patch

```diff
@@ -0,0 +1,176 @@
+From 25e012a5ebed83930e84c7c8cf0cd0e0b4e84dee Mon Sep 17 00:00:00 2001
+From: Sam Meluch <sammeluch@microsoft.com>
+Date: Tue, 14 May 2024 17:03:13 -0700
+Subject: [PATCH] Skip tests for mariner
+
+---
+ tests/test_card.py                   | 11 +++++++++++
+ tests/test_markdown.py               | 11 +++++++++++
+ tests/test_markdown_no_hyperlinks.py | 11 +++++++++++
+ tests/test_syntax.py                 |  8 +++++++-
+ 4 files changed, 40 insertions(+), 1 deletion(-)
+
+diff --git a/tests/test_card.py b/tests/test_card.py
+index d578ec89..f1683af2 100644
+--- a/tests/test_card.py
++++ b/tests/test_card.py
+@@ -1,11 +1,21 @@
+ import io
+ import re
++import sys
++import pytest
+ 
+ from rich.__main__ import make_test_card
+ from rich.console import Console, RenderableType
+ 
+ from ._card_render import expected
+ 
++if sys.version_info >= (3, 8):
++    from importlib.metadata import Distribution
++else:
++    from importlib_metadata import Distribution
++
++PYGMENTS_VERSION = Distribution.from_name("pygments").version
++OLD_PYGMENTS = PYGMENTS_VERSION == "2.5.2"
```

Comment:
I'm guessing we won't go down with `python-pygments` but does this mean that updating to `python-pygments` version 2.5.3 or 2.6.0 will already fix this issue? And if so, can we try the upgrade instead of the patch? Patches, which don't have a good chance of disappearing with version upgrades (like CVE patches for instance) add an additional maintenance burden, since they tend to break and require updates themselves.

Reply:
I talked with Andrew about upgrading the package. The required upgrade is to >=2.14.0. I started down the avenue of upgrading to latest (2.18.0) and found the package needed many additional dependencies as a part of the upgrade. While this is not normally an issue, since 'python-pygments' is a part of the toolchain, these dependencies would also need to be added as such there. Instead we decided it is best to simply skip these tests for the time being as they are mainly just comparing loaded style options from the library anyway.

I left a comment in the spec file detailing this patch's removal after such an upgrade.